templates/vue/src/components/tyde/TydeCockpit.vue

```diff
@@ -105,12 +100,45 @@ export default {
         return ""
       }
     },
+    moduleStyles(node) {
+      return {
+        backgroundImage: `url(${node.typeData.spaceshipPartNotEarnedIconUrl})`,
+        top: node.typeData.spaceshipPartY+'px',
+        left: node.typeData.spaceshipPartX+'px',
+        height: node.typeData.spaceshipPartHeight+'px',
+        width: node.typeData.spaceshipPartWidth+'px'
+      }
+    },
+    moduleImage(node) {
+      if (node.tydeProgress == 1) {
+        return node.typeData.spaceshipPartEarnedIconUrl
+      } else {
+        return node.typeData.spaceshipPartNotEarnedIconUrl
+      }
+    },
+    isNodeModuleType(node) {
+      return node.tydeType === tydeTypes.MODULE
+    },
+    nodeMouseOverHandler(node) {
+      document.getElementById("tyde-cockpit-"+node.id).style.backgroundImage = `url(${node.typeData.spaceShipPartHoverIconUrl})`
+    },
+    nodeMouseLeaveHandler(node) {
+      document.getElementById("tyde-cockpit-"+node.id).style.backgroundImage = `url(${this.moduleImage(node)})`
+    },
```

Comment:
This is not the right way of doing things in Vue. I recommend creating a component for a spaceship part. Inside that component, keep a state to determine which image should be used. Update this state when the user hovers over or away, and tie the background image to the state. Let me know if you want to chat about this to either myself or Narendra.